test/widgets/text_test.dart

```diff
@@ -96,11 +95,13 @@ void main() {
       required FontWeight expectedFontWeight,
     }) async {
       testWidgets(description, (WidgetTester tester) async {
-        await tester.pumpWidget(
-          MaterialApp(
-            home: MediaQuery(
-              data: MediaQueryData(boldText: platformRequestsBold),
-              child: Builder(builder: (context) => Text('', style: styleBuilder(context))))));
+        addTearDown(testBinding.reset);
+        tester.platformDispatcher.accessibilityFeaturesTestValue =
+          FakeAccessibilityFeatures(boldText: platformRequestsBold);
+        addTearDown(tester.platformDispatcher.clearAccessibilityFeaturesTestValue);
```

Comment:
This seems fine to include in the same commit, though. It's fairly straightforwardly just another way of expressing the effect of that `MediaQuery`.